compiler/resolution/functionResolution.cpp

```diff
@@ -4228,6 +4228,21 @@ static bool isInitEqualsPreResolve(CallExpr* call) {
   return result;
 }
 
+static bool obviousMisMatch(CallExpr* call, FnSymbol* fn) {
+  bool isMethod = isMethodPreResolve(call);
```

Comment:
I think this function would be more clearly written

```
bool isMethodCall = isMethodPreResolve(call);
bool isMethodFn = (fn->_this != NULL);
return (isMethodCall != isMethodFn);
```

Reply:
Yeah, I agree (and realized this after I requested the review, but figured I'd save all the changes for one pass).  I wrote it this way originally because (a) I didn't think it would be this simple and (b) I had aspirations to analyze the class hierarchies that a method call was involved in to eliminate unrelated class hierarchies.  But then I was wrong on the first and didn't bother with the second.